.github/workflows/publish.yml

```diff
@@ -0,0 +1,30 @@
+name: Upload Python Package
```

Comment:
It may be worth adding a comment pointing to the information in pyproject.toml.

```suggestion
# For more information, see the "Release a new version" notes in pyproject.toml.

name: Upload Python Package
```